graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ServiceTypesGeneratorTest.java

```diff
@@ -0,0 +1,624 @@
+/*
+ *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.graphql.generator.ballerina;
+
+import graphql.schema.GraphQLSchema;
+import io.ballerina.graphql.common.GraphqlTest;
+import io.ballerina.graphql.common.TestUtils;
+import io.ballerina.graphql.exception.ValidationException;
+import io.ballerina.graphql.generator.service.GraphqlServiceProject;
+import io.ballerina.graphql.generator.service.exception.ServiceTypesGenerationException;
+import io.ballerina.graphql.generator.service.generator.ServiceTypesGenerator;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static io.ballerina.graphql.common.TestUtils.writeContentTo;
+
+/**
+ * Test class for ServiceTypesGenerator.
+ * Test the successful generation of service types file code
+ */
+public class ServiceTypesGeneratorTest extends GraphqlTest {
+    @Test(description = "Test for schema with basic, method - default")
```

Comment:
Instead of defining these in the description, shall we add them to proper [test groups](https://www.tutorialspoint.com/testng/testng_group_test.htm)? That way, we can filter the tests when needed.

Maybe use more explanatory names. `basic` and `default` does not covey the idea right away.

Reply:
fixed it. I added test groups and changed the names